File: src/_/types/role.ts
export type RoleSummary = {
  id: number;
  name: string;
  description: string;
  user_count: number;
  created_at: string;
};

export type CreateRoleRequest = {
  name: string;
  description?: string;
};

export type UpdateRoleRequest = {
  name: string;
  description?: string;
};

File: src-tauri/src/models/role.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct RoleSummary {
    pub id: i32,
    pub name: String,
    pub description: String,
    pub user_count: i64,
    pub created_at: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateRoleRequest {
    pub name: String,
    pub description: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateRoleRequest {
    pub name: String,
    pub description: Option<String>,
}
